nordic: nrf54h20: add support for SPIS120

nika-nordic opened the pull request:
PR adds support for SPIS120 instance in dts and `spi` driver.

nordic-krch commented on tests/drivers/spi/spi_controller_peripheral/testcase.yaml:
```diff
@@ -64,6 +64,16 @@ tests:
+    # SPIS120 instance occupies P6 which is not available on nRF54H20 DK pin headers
+    build_only: true
+    extra_args: DTC_OVERLAY_FILE="boards/nrf54h20dk_nrf54h20_cpuapp_fast_spis.overlay"
+    platform_exclude:
```
maybe platform_allow would be better?

nordic-krch commented on tests/drivers/spi/spi_controller_peripheral/testcase.yaml:
```diff
@@ -64,6 +64,16 @@ tests:
       - nrf54h20dk/nrf54h20/cpuppr
 
+  drivers.spi.spis_fast:
+    # SPIS120 instance occupies P6 which is not available on nRF54H20 DK pin headers
```
Maybe extend the comment with information that configuration is added only for compilation check. I know that there is` `build_only:true` but it might not be obvious that it cannot be run.

nordic-krch reviewed (commented):
looks ok.

nika-nordic committed:
soc: nordic: add support for SPIS120 instance

It is defined as spis120 rather than spi120,
because spi120 is already used for SPIM120 hardware instance,
but their base address is different.

Signed-off-by: Nikodem Kastelik <nikodem.kastelik@nordicsemi.no>

nika-nordic committed:
drivers: spi: nrf: add support for SPIS120 instance

SPIS120 is a SPI slave device with >8 MHz SCK compatibility.

Signed-off-by: Nikodem Kastelik <nikodem.kastelik@nordicsemi.no>

nika-nordic committed:
tests: drivers: spi: add test case for nRF54H20 SPIS120

New test case is for building an application simultaneously
utilizing SPIM120, SPIM120, SPIS120 and SPIS13x.

Signed-off-by: Nikodem Kastelik <nikodem.kastelik@nordicsemi.no>

nika-nordic commented:
closing in favor of already merged: https://github.com/zephyrproject-rtos/zephyr/pull/86968
